embassy-rp/src/multicore.rs

```diff
@@ -255,23 +264,55 @@ where
 
     // Wait until the other core has copied `entry` before returning.
     fifo_read();
+
+    // Enable FIFO interrupts from core1.
+    #[cfg(feature = "rp2040")]
+    unsafe {
+        interrupt::SIO_IRQ_PROC0.enable()
+    };
+    #[cfg(feature = "_rp235x")]
+    unsafe {
+        interrupt::SIO_IRQ_FIFO.enable()
+    };
 }
 
 /// Pause execution on CORE1.
 pub fn pause_core1() {
     if IS_CORE1_INIT.load(Ordering::Acquire) {
+        cortex_m::interrupt::disable();
         fifo_write(PAUSE_TOKEN);
         // Wait for CORE1 to signal it has paused execution.
-        while fifo_read() != PAUSE_TOKEN {}
+        loop {
+            let token = fifo_read();
+            if token == PAUSE_TOKEN {
+                break;
+            }
+            #[cfg(feature = "fifo-handler")]
+            handle_fifo_token(token);
+        }
+        unsafe {
+            cortex_m::interrupt::enable();
+        }
     }
 }
 
 /// Resume CORE1 execution.
 pub fn resume_core1() {
     if IS_CORE1_INIT.load(Ordering::Acquire) {
+        cortex_m::interrupt::disable();
```

Comment:
so this is wrapped in disable/enable interrupts. but when interrupts were disabled when entering this function, they are unconditionally enabled after. not sure this is acceptable ... e.g., if this was called from within a critical section, there is an issue.
better enclose in `cortex_m:m::interrupt::free()`?

Reply:
Another good catch, thanks! Done.